acceptance-tests/cloudsql_pg_migration_test.go

```diff
@@ -0,0 +1,176 @@
+				"postgres_version":      "POSTGRES_11",
+				"db_name":               databaseName,
+				"public_ip":             false,
+				"backups_retain_number": 0,
+			}),
+			services.WithAsync(),
+		)
+
+		By("creating the original service instance")
+		sourceServiceOffering := "google-cloudsql-postgres-vpc"
+		sourceServicePlan := "default"
+		sourceServiceInstance := services.CreateInstance(
+			sourceServiceOffering,
+			sourceServicePlan,
+			services.WithBroker(&brokers.Broker{Name: legacyBrokerName}),
+			services.WithParameters(map[string]any{
+				"tier":            legacyDBTier,
+				"private_network": os.Getenv("GCP_PAS_NETWORK"),
+				"database_name":   databaseName,
+				"backups_enabled": "false",
+			}),
+		)
+		defer sourceServiceInstance.Delete()
+
+		By("binding an app to the source service instance")
+		sourceApp := apps.Push(apps.WithApp(apps.PostgreSQL))
+		defer sourceApp.Delete()
+
+		sourceInstanceBinding := sourceServiceInstance.Bind(sourceApp)
+		sourceApp.Start()
+		defer sourceApp.DELETETestTable()
+
+		By("creating a schema and adding some data in the source database")
+		schema := random.Name(random.WithMaxLength(8))
+		sourceApp.PUT("", schema)
+		defer sourceApp.DELETE(schema)
+
+		key := random.Hexadecimal()
+		value := random.Hexadecimal()
+		sourceApp.PUT(value, "%s/%s", schema, key)
+
+		By("waiting for the new service creation to succeed")
+		services.WaitForInstanceCreation(targetServiceInstance.Name)
+		defer targetServiceInstance.Delete()
+
+		By("extracting a service key for the legacy service instance")
+		credentials := sourceInstanceBinding.Credential()
+		legacyBinding, err := legacybindings.ExtractPostgresBinding(credentials)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a bucket")
+		bucketName := "bucket-" + databaseName
+		gsql.CreateBackupBucket(bucketName)
+		defer gsql.DeleteBucket(bucketName)
+
+		By("creating extensions in the source db")
+		gsql.PerformAdminSQL(createExtensionSQL, legacyBinding.InstanceName, legacyBinding.DatabaseName, bucketName)
+
+		By("performing the backup")
+		backupURI := gsql.CreateBackup(legacyBinding.InstanceName, databaseName, bucketName)
+
+		By("preparing the restore in the target instance")
+		targetInstanceIaaSName := fmt.Sprintf("csb-postgres-%v", targetServiceInstance.GUID())
+		gsql.PerformAdminSQL(createBindingUserGroupSQL, targetInstanceIaaSName, databaseName, bucketName)
+
+		By("restoring the backup onto the new service instance")
+		gsql.RestoreBackup(backupURI, targetInstanceIaaSName, databaseName)
+
+		By("cleaning up after the restore")
+		gsql.PerformAdminSQL(disableBindingUserGroupLoginSQL, targetInstanceIaaSName, databaseName, bucketName)
+
+		targetApp := apps.Push(apps.WithApp(apps.PostgreSQL))
+		defer targetApp.Delete()
+		targetBinding := targetServiceInstance.Bind(targetApp)
+		targetApp.Start()
+
+		By("reading the data")
+		got := targetApp.GET("%s/%s", schema, key)
+		Expect(got).To(Equal(value))
+
+		By("creating a new schema")
```

Comment:
It's not immediately obvious why the test continues once the data has been read back. Could some messaging be added, for example `By("creating a new schema to test that ....")`